xbmc/addons/AddonManager.cpp

```diff
@@ -247,6 +247,35 @@ std::vector<std::shared_ptr<IAddon>> CAddonMgr::GetAvailableUpdatesOrOutdatedAdd
   return result;
 }
 
+bool CAddonMgr::GetAvailableUpdatesAndOutdatedAddons(
+    std::set<std::shared_ptr<IAddon>>& outdated, std::set<std::shared_ptr<IAddon>>& updates) const
+{
+  CSingleLock lock(m_critSection);
+  auto start = XbmcThreads::SystemClockMillis();
+
+  std::vector<std::shared_ptr<IAddon>> result;
+  std::vector<std::shared_ptr<IAddon>> installed;
+  CAddonRepos addonRepos(*this);
+
+  addonRepos.LoadAddonsFromDatabase(m_database);
+
+  GetAddonsForUpdate(installed);
+
+  addonRepos.BuildOutdatedList(installed, result);
+  std::transform(result.begin(), result.end(), std::inserter(outdated, outdated.end()),
+                 [](std::shared_ptr<IAddon>& addon) { return addon; });
+
+  result.clear();
+  addonRepos.BuildUpdateList(installed, result);
+  std::transform(result.begin(), result.end(), std::inserter(updates, updates.end()),
+                 [](std::shared_ptr<IAddon>& addon) { return addon; });
```

Comment:
tried to use:
```cpp
  std::transform(make_move_iterator(result.begin()),
                 make_move_iterator(result.end()),
  std::inserter(updates, updates.end()),
  [](std::shared_ptr<IAddon>&& addon) { return std::move(addon); });
```
to get rid of `result.clear()` but that crashed very badly on a testrun.
has anyone an idea what's wrong with this?

Reply:
switch to `std::vector` as use of `std::set` doesn't make sense here anymore